src/puppetlabs/puppetdb/config.clj

```diff
@@ -153,12 +154,23 @@
    :subprotocol "hsqldb"
    :subname     (format "file:%s;hsqldb.tx=mvcc;sql.syntax_pgs=true" (io/file vardir "db"))})
 
-(defn defaulted-db-connection
-  "Adds each of `hsql-default-connection` keypairs to `database` if classname, subprotocol and subname
-   are not found in the `database` config"
+(defn validate-db-settings
+  "Throws a {:type ::configuration-error :message m} exception
+  describing the required additions if the [database] configuration
+  doesn't specify classname, subprotocol and subname, all of which are
+  now required."
   [{:keys [global database] :as config :or {database {}}}]
```

Comment:
so, 'database' here is a config map? we should rename it to 'database-config' or something if that's the case. As it is, `(when (not-every? database [...]))` makes zero sense.